handler_test.go

```diff
@@ -0,0 +1,1982 @@
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionFalse,
+								Reason:  dns.RcodeToString[dns.RcodeSuccess],
+								Message: rcodeMessage[dns.RcodeSuccess],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Update only wildcard dns name resolver object status with wildcard dns name",
+		dnsNameResolvers: []ocpnetworkapiv1alpha1.DNSNameResolver{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "regular",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "www.example.com.",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcard",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "*.example.com.",
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "*.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("*.example.com. 30 IN A 1.1.1.1"),
+					test.A("*.example.com. 30 IN A 1.1.1.2"),
+				},
+			},
+		},
+		expectedStatuses: []ocpnetworkapiv1alpha1.DNSNameResolverStatus{
+			{},
+			{
+				ResolvedNames: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedName{
+					{
+						DNSName: "*.example.com.",
+						ResolvedAddresses: []ocpnetworkapiv1alpha1.DNSNameResolverResolvedAddress{
+							{
+								IP:         "1.1.1.1",
+								TTLSeconds: 30,
+							},
+							{
+								IP:         "1.1.1.2",
+								TTLSeconds: 30,
+							},
+						},
+						ResolutionFailures: 0,
+						Conditions: []metav1.Condition{
+							{
+								Type:    ConditionDegraded,
+								Status:  metav1.ConditionFalse,
+								Reason:  dns.RcodeToString[dns.RcodeSuccess],
+								Message: rcodeMessage[dns.RcodeSuccess],
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		name: "Update regular dns name resolver object status with regular dns name and wildcard dns name resolver object status with wildcard dns name",
+		dnsNameResolvers: []ocpnetworkapiv1alpha1.DNSNameResolver{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "regular",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "www.example.com.",
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "wildcard",
+					Namespace: "dns",
+				},
+				Spec: ocpnetworkapiv1alpha1.DNSNameResolverSpec{
+					Name: "*.example.com.",
+				},
+			},
+		},
+		dnsQueryTestCases: []test.Case{
+			{
+				Qname: "*.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("*.example.com. 30 IN A 1.1.1.1"),
+					test.A("*.example.com. 30 IN A 1.1.1.2"),
+				},
+			},
+			{
+				Qname: "www.example.com.",
+				Qtype: dns.TypeA,
+				Rcode: dns.RcodeSuccess,
+				Answer: []dns.RR{
+					test.A("www.example.com. 30 IN A 1.1.1.1"),
+					test.A("www.example.com. 30 IN A 1.1.1.2"),
```

Comment:
I think it's better to make the IP addresses different here to make sure they were not mixed up by the implementation.

Reply:
The IPs are made same to demonstrate that though the details of the regular DNS name is not added to the status of the DNSNameResolver corresponding to the wildcard DNS name, but added to the status of the DNSNameResolver corresponding to the regular DNS name.

This is in line with the previous comment (https://github.com/openshift/coredns-ocp-dnsnameresolver/pull/1#discussion_r1526232810) and tests the behavior for a DNSNameResolver object for a regular DNS name.

Reply:
For the sake of completeness of the behavior: can we have a test case where wildcard and regular name matching the wildcard would have different addresses?

Reply:
This test case is already added here: https://github.com/openshift/coredns-ocp-dnsnameresolver/pull/1/files#diff-39de3493ae7887b6c07ddf65fc59c005b3f2fccc9b06da022954597de5b12d7cR1192-R1275